### ISSUE TITLE
[bug] <the modal cannot close>

### ISSUE THREAD

**Issue by wwei-github** [rainbow-me/rainbowkit#837]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

0.7.2

### wagmi Version

0.7.5

### Current Behavior

Using the official example, click connectbutton to connect to the matemask wallet, and when it's done, click disconnect, and the X in the popover won't close the popover
 

### Expected Behavior

I hope it can cancel the modal

### Steps To Reproduce

click connectButton -> matemask -> disconnect -> click modal X

**Comment by rtman** [comment]
Also encounter this occasionally.

**Comment by mathesond2** [comment]
+1 I've got the same issue currently, happy to speak more to it

**Comment by jacobedawson** [comment]
Seems to be related to this error - there's some issue with changing the state on the unmounted modals:

![image](https://user-images.githubusercontent.com/4788506/198268991-90c9c42d-dedd-4ad2-b06c-5c3557d99f4a.png)

**Comment by actuallymentor** [comment]
Can confirm the same issue, also same error as @jacobedawson.

**Comment by cattingcat** [comment]
I have the same issue, but I have error log:
```
react-remove-scroll-bar: cannot calculate scrollbar size because it is removed (overflow:hidden on body 
```

My hypothesis is that:
When you open connect-wallet dialog - scroll on body really becomes hidden, 
After that you are clicking connect with metamask
Old dialog closes and new dialog with wallet details opens, and it is trying to hide scrollbar again, but library for hiding wallet throws exception because scrollbar already hidden

https://github.com/theKashey/react-remove-scroll-bar/blob/3dd80e28c92b8aac025e41f4258ff926cdc88af9/src/utils.ts#L22

**Comment by Tingwei-Justin** [comment]
Same issue. Hope can get a quick fix

**Comment by wassienabe** [comment]
Same issue here, not seeing any errors however, the close button simply doesn't respond after disconnecting elsewhere.

**Comment by outdoteth** [comment]
+1. For me, the "disconnect" and "copy address" buttons are responsive, it's only the "x" that doesnt respond.

**Comment by jacobedawson** [comment]
@wassienabe @outdoteth @Tingwei-Justin @cattingcat @actuallymentor found the issue / quick fix: https://github.com/rainbow-me/rainbowkit/issues/836#issuecomment-1303571558

**Comment by sarinniural** [comment]
I am seeing the same issue.

**Comment by dujomaton** [comment]
+1 on this. "Disconnect" and "Copy Address" work fine, just can't close.

**Comment by mrruby** [comment]
Still can't close, even with the new version 0.8.0

**Comment by spobwoode** [comment]
+1

**Comment by morph-dev** [comment]
+1

**Comment by Giulianomlodi** [comment]
I've the same issue. Thanks in advance for the fix

**Comment by Giulianomlodi** [comment]
From what I understand, the issue relates to the user popup not being able to be closed after logging in with RainbowKit. This issue only occurs when the user attempts to close the user popup immediately after login - if the page is refreshed and the user popup is opened manually, the close icon works correctly.

**Comment by BackToTheCode** [comment]
+1

**Comment by Ataye** [comment]
Happening for me on 0.8.1, no errors logged to console.

Disconnect and refresh gets re-connected (autoConnect=true).  Disconnect then reconnect and the popup (with tree and wallet address etc) cannot be closed via X or background click.

**Comment by nestorolivaresh** [comment]
> Happening for me on 0.8.1, no errors logged to console.
> 
> Disconnect and refresh gets re-connected (autoConnect=true). Disconnect then reconnect and the popup (with tree and wallet address etc) cannot be closed via X or background click.
> 
> And indeed, this seems to be related to StrictMode, so i don't know if its a bug or a StrictMode development feature :)

Same for me. 0.8.1 version, using `ConnectButton.Custom` and StrictMode in false. Any solutions to this?

**Comment by advra** [comment]
Bump. Same issues still with trying to close. Buttons inside work but closing X or clicking outside the model to attempt closing the modal will not close.

**Comment by DanielSinclair** [comment]
This issue has been resolved with wagmi `0.11.4` [here](https://github.com/wagmi-dev/wagmi/releases/tag/wagmi%400.11.4).
There are no conflicts to bump to this version with rainbowkit `0.8.1`, but `0.11.x` will be released tomorrow anyway for parity: #1000
